contracts/src/v0.8/vrf/dev/VRFCoordinatorV2_5.sol

```diff
@@ -625,19 +628,9 @@ contract VRFCoordinatorV2_5 is VRF, SubscriptionAPI, IVRFCoordinatorV2Plus {
     if (consumersLength == 0) {
       return false;
     }
-    uint256 provingKeyHashesLength = s_provingKeyHashes.length;
     for (uint256 i = 0; i < consumersLength; ++i) {
-      address consumer = consumers[i];
-      for (uint256 j = 0; j < provingKeyHashesLength; ++j) {
-        (uint256 reqId, ) = _computeRequestId(
-          s_provingKeyHashes[j],
-          consumer,
-          subId,
-          s_consumers[consumer][subId].nonce
-        );
-        if (s_requestCommitments[reqId] != 0) {
-          return true;
-        }
+      if (s_consumers[consumers[i]][subId].pendingReqCount > 0) {
```

Comment:
Currently both removeConsumer and cancelSubscription call pendingRequestExists, which checks if a subId has pendingRequests or not.

This means we would not be able to delete a consumer of a subId if another consumer of the same subId has pendingRequests. Is this a limiting factor?

Reply:
The pendingRequestsExists method tracks pending requests on an entire subscriptionID when we check to removeConsumer.
Shall we add a new method that just checks s_consumers[address][subId].pendingReqCounter when removeConsumer is called

Reply:
Discussed this offline. We will leave as is because customers only removeConsumer when they want to cancel subscriptions anyways.